internal/controller/streams.go

```diff
@@ -389,11 +394,7 @@ func (sm *streamManager) StreamFalcoNetworkFlows(ctx context.Context, logger *za
 				logger.Error("Failed to parse Falco event into flow", zap.Error(err))
 				return err
 			}
-			err = sm.sendNetworkFlowRequest(logger, convertedFalcoFlow)
-			if err != nil {
-				logger.Error("Failed to send Falco flow", zap.Error(err))
-				return err
-			}
+			sm.flowChannel <- convertedFalcoFlow
```

Comment:
Never do a blocking write into a channel ignoring `ctx`.
Just wrap that into a method of `FlowCache` as I suggested above.
```suggestion
			sm.flowCache.CacheFlow(ctx, convertedFalcoFlow)
```